catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java

```diff
@@ -550,6 +572,22 @@ public SolrQuery propertyIsLessThanOrEqualTo(String propertyName, double literal
         return getLessThanOrEqualToQuery(propertyName, AttributeFormat.DOUBLE, literal);
     }
 
+    public SolrQuery propertyIsDivisibleBy(String propertyName, long divisor) {
+        //use the sort key for the field since divisible can't operate on multivalued fields (it will always be a single value).
+        List<String> solrExpressions = resolver.getAnonymousField(propertyName)
+                .stream()
+                .map(resolver::getSortKey)
+                .map(f -> "_val_:\"{!frange l=0 u=0}mod(" + f + "," + divisor + ")\"")
```

Comment:
✏️ 1. Favor `String.format()` over concatenation.
2. I think this is probably a complex enough lambda that it distracts from the form of the stream logic. It would read better if you extracted a method to convert the `sortKey` to this String representation and then in your map just called that method.